app/controllers/api/v1/auth_controller.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class AuthController < ApplicationController
+      before_action :user, only: [:create]
+      class AuthenticationError < StandardError; end
+
+      rescue_from AuthenticationError, with: :handle_unauthenticated
+
+      # POST api/v1/auth/login
+      def create
+        if @user
+          raise AuthenticationError unless auth_user
+
+          render json: @user.as_json(only: %i[email first_name last_name role_id])
+        else
+          render json: { error: 'No such user' }, status: :unauthorized
+        end
+      end
+
+      private
+
+      def auth_user
+        @user.authenticate(params.require(:password))
+      end
+
+      def handle_unauthenticated
+        render json: { ok: false }
+      end
+
+      def user
+        @user ||= User.find_by(email: params.require(:email))
```

Comment:
Teniendo el `exception handler`, puede usar find para arrojar una excepción en caso de que el usuario no exista.
```suggestion
        @user ||= User.find(email: params.require(:email))
```

Reply:
No queda claro, porque se sugiere modificar `User.find` (solo busca por el valor de ID) por `User.find_by`.